common/djangoapps/third_party_auth/tests/test_models.py

```diff
@@ -53,3 +58,91 @@ def test_clean_username_unicode_enabled(self):
         Test the username cleaner function with unicode enabled
         """
         assert clean_username('ItJüstWòrks™') == 'ItJüstWòrks'
+
+    def test_saml_configuration_dynamic_lookup(self):
+        """
+        Test SAML configuration lookup functionality, including the toggle behavior and error handling.
+        """
+        site = SiteFactory()
+
+        # Create initial SAML configuration
```

Comment:
Initial setup should be defined inside `setUp(self)` function. That'll clean up the test case and make it easier to review the changes.